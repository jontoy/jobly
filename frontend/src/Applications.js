import React, {useContext} from 'react';
import JobsList from './JobsList';
import UserContext from './userContext';
import useApply from './hooks/useApply';

const Applications = () => {
    const {currentUser} = useContext(UserContext);
    const [apply, unapply] = useApply();

    return (
        <div className="Applications container mt-4">
            <h2 className="display-5">Current Applications</h2>
            <JobsList jobs={currentUser.jobs} apply={apply} unapply={unapply}/>
        </div>
    )
}

export default Applications;