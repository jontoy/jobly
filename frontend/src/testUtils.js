import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
