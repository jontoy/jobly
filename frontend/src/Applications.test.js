import React from 'react';
import { render } from '@testing-library/react';
import Applications from './Applications';
import {UserProvider} from './testUtils';


it('renders without crashing', () => {
    render(
    <UserProvider>
        <Applications/>
    </UserProvider>
    );
});
  
it('matches snapshot', () => {
      const {asFragment} = render(
      <UserProvider>
        <Applications/>
      </UserProvider>);
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when user has existing applications', () => {
    const currentUser = {username:'test user', jobs:[{id:1, title:'fake job'}]}
    const {asFragment} = render(
    <UserProvider appliedIds={[1]} currentUser={currentUser}>
        <Applications/>
    </UserProvider>
    );
  expect(asFragment()).toMatchSnapshot();
});

