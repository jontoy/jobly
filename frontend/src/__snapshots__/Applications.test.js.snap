// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Applications container mt-4"
  >
    <h2
      class="display-5"
    >
      Current Applications
    </h2>
    <ul
      class="JobsList list-unstyled"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user has existing applications 1`] = `
<DocumentFragment>
  <div
    class="Applications container mt-4"
  >
    <h2
      class="display-5"
    >
      Current Applications
    </h2>
    <ul
      class="JobsList list-unstyled"
    >
      <li>
        <div
          class="CompanyCard card p-2 mb-1"
        >
          <h5
            class="card-title"
          >
            fake job
          </h5>
          <p
            class="card-text"
          >
            <br />
          </p>
          <button
            class="btn btn-danger ml-auto"
          >
            Remove Application
          </button>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
